### PR TITLE
Require `[compat]` entry for `julia`

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves https://github.com/JuliaTesting/Aqua.jl/issues/235.

As this is point 7 in the [guidelines for the General registry](https://juliaregistries.github.io/RegistryCI.jl/stable/guidelines/#New-packages), I support Aqua checking this.

### CODE REVIEW

**Comment by lgoettgens** [comment]
@fingolfin Do you wanna have a look at this? If not, I will just merge this and https://github.com/JuliaTesting/Aqua.jl/pull/223, and then release 0.8.0.

File: src/deps_compat.jl [hunk 1]
```diff
@@ -23,18 +24,29 @@ to the corresponding `check_\$x` keyword argument.
...
 """
-function test_deps_compat(pkg::PkgId; check_extras = true, check_weakdeps = true, kwargs...)
+function test_deps_compat(
+    pkg::PkgId;
+    check_julia = true,
+    check_extras = true,
+    check_weakdeps = true,
+    kwargs...,
+)
+    if check_julia !== false
```
Comment (hyrodium) [inline src/deps_compat.jl:34]:
```suggestion
    check_julia::Bool = true,
    check_extras::Bool = true,
    check_weakdeps::Bool = true,
    kwargs...,
)
    if check_julia
```

I think argument-type specification is better for readability and error handling.

**Review by hyrodium — COMMENTED** [review]
Added a minor comment. Other changes LGTM.

File: src/deps_compat.jl (hunk 1 above)
Comment (lgoettgens) [reply src/deps_compat.jl:34]: That is not possible here in an easy way, as one can pass a `Bool` or a `NamedTuple`. The docstring says `A NamedTuple can be used to pass keyword arguments with test options (see below).`

**Review by hyrodium — APPROVED** [review]

**Pull request merged**